### DISCUSSION TITLE
coding projects

### DISCUSSION THREAD

**Discussion by Samarthmalpani** [Heapy/awesome-kotlin#678] [Ideas]
hello all, I am currently learning kotlin, and c++ which means I am at an intermediate level. I want to build something in coding but I am only getting ideas that are already done by someone can anyone plz tell me a new coding idea that will different and give me a new learning.

**Comment by IRus** [comment]
You can always learn even solving already solved problems, like writing compiler/interpretator, db, http client/server, etc
Because you'll find all not so obvious design decisions (trade-offs) that developers of this software made, even better if you will compare your solution with existing one. There are always interesting problems around asynchronous/concurrent code and performance.
